cpp/src/arrow/compute/kernels/vector_array_sort.cc

```diff
@@ -590,6 +590,13 @@ void AddArraySortingKernels(VectorKernel base, VectorFunction* func) {
     base.exec = GenerateNumeric<ExecTemplate, UInt64Type>(*physical_type);
     DCHECK_OK(func->AddKernel(base));
   }
+  {
+    // XXX: float16() is not in NumericTypes() yet
+    auto physical_type = GetPhysicalType(float16());
```

Comment:
What is the underlying physical type btw? We should be careful that float16 NaNs sort as described as https://arrow.apache.org/docs/cpp/compute.html#sorts-and-partitions, and we should test for that.

Reply:
It's `float16` itself. I'm changing the code to make that more obvious.